File: go/src/leetcode/lc1342.go
package leetcode

func numberOfSteps(num int) int {
	if num <= 0 {
		return 0
	}
	steps := 0
	for num > 0 {
		steps++
		if (num & 1) == 1 {
			steps++
		}
		num >>= 1
	}
	return steps - 1
}

File: js/leetcode/lc1342.js
var numberOfSteps = function (num) {
    if (num <= 0) return 0;
    let steps = 0;
    do {
        steps += (num & 1) == 0 ? 1 : 2;
        num >>= 1;
    } while (num);
    return --steps;
};

module.exports = {
    numberOfSteps: numberOfSteps
}